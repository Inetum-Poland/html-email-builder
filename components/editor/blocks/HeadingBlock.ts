import type { BlockProperties } from "grapesjs";
import {
  Heading1,
  Heading2,
  Heading3,
  Heading4,
  Heading5,
  Heading6,
} from "lucide-static";

const icons = new Map([
  [1, Heading1],
  [2, Heading2],
  [3, Heading3],
  [4, Heading4],
  [5, Heading5],
  [6, Heading6],
]);

const sizes = new Map([
  [1, 2],
  [2, 1.5],
  [3, 1.17],
  [4, 1],
  [5, 0.83],
  [6, 0.67],
]);

export const HeadingBlock = (level: 1 | 2 | 3 | 4 | 5 | 6): BlockProperties => {
  const { $i18n: { t } } = useNuxtApp();
  const fontSize = sizes.get(level) || 1;
  const lineHeight = 1.15;

  return {
    label: `H${level}`,
    media: icons.get(level),
    content: `
    <mj-text font-size="${fontSize}rem" line-height="${lineHeight}">
      ${t("heading")}
    </mj-text>`,
    activate: true,
    attributes: { title: `${t("heading")} ${level}` },
    category: t("genericBlocks"),
  };
};
