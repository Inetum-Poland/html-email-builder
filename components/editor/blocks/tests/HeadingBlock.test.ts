import { expect, test } from "vitest";
import { HeadingBlock } from "../HeadingBlock";

test.each([1, 2, 3, 4, 5, 6] as const)("renders the H%i heading", (level) =>
  expect(HeadingBlock(level)).toMatchSnapshot()
);
