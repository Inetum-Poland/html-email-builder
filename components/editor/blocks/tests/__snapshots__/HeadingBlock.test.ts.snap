// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the H1 heading 1`] = `
{
  "activate": true,
  "attributes": {
    "title": "Heading 1",
  },
  "category": "Generic blocks",
  "content": "
    <mj-text font-size="2rem" line-height="1.15">
      Heading
    </mj-text>",
  "label": "H1",
  "media": "
<svg
  class="lucide lucide-heading-1"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 12h8" />
  <path d="M4 18V6" />
  <path d="M12 18V6" />
  <path d="m17 12 3-2v8" />
</svg>
",
}
`;

exports[`renders the H2 heading 1`] = `
{
  "activate": true,
  "attributes": {
    "title": "Heading 2",
  },
  "category": "Generic blocks",
  "content": "
    <mj-text font-size="1.5rem" line-height="1.15">
      Heading
    </mj-text>",
  "label": "H2",
  "media": "
<svg
  class="lucide lucide-heading-2"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 12h8" />
  <path d="M4 18V6" />
  <path d="M12 18V6" />
  <path d="M21 18h-4c0-4 4-3 4-6 0-1.5-2-2.5-4-1" />
</svg>
",
}
`;

exports[`renders the H3 heading 1`] = `
{
  "activate": true,
  "attributes": {
    "title": "Heading 3",
  },
  "category": "Generic blocks",
  "content": "
    <mj-text font-size="1.17rem" line-height="1.15">
      Heading
    </mj-text>",
  "label": "H3",
  "media": "
<svg
  class="lucide lucide-heading-3"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 12h8" />
  <path d="M4 18V6" />
  <path d="M12 18V6" />
  <path d="M17.5 10.5c1.7-1 3.5 0 3.5 1.5a2 2 0 0 1-2 2" />
  <path d="M17 17.5c2 1.5 4 .3 4-1.5a2 2 0 0 0-2-2" />
</svg>
",
}
`;

exports[`renders the H4 heading 1`] = `
{
  "activate": true,
  "attributes": {
    "title": "Heading 4",
  },
  "category": "Generic blocks",
  "content": "
    <mj-text font-size="1rem" line-height="1.15">
      Heading
    </mj-text>",
  "label": "H4",
  "media": "
<svg
  class="lucide lucide-heading-4"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 12h8" />
  <path d="M4 18V6" />
  <path d="M12 18V6" />
  <path d="M17 10v4h4" />
  <path d="M21 10v8" />
</svg>
",
}
`;

exports[`renders the H5 heading 1`] = `
{
  "activate": true,
  "attributes": {
    "title": "Heading 5",
  },
  "category": "Generic blocks",
  "content": "
    <mj-text font-size="0.83rem" line-height="1.15">
      Heading
    </mj-text>",
  "label": "H5",
  "media": "
<svg
  class="lucide lucide-heading-5"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 12h8" />
  <path d="M4 18V6" />
  <path d="M12 18V6" />
  <path d="M17 13v-3h4" />
  <path d="M17 17.7c.4.2.8.3 1.3.3 1.5 0 2.7-1.1 2.7-2.5S19.8 13 18.3 13H17" />
</svg>
",
}
`;

exports[`renders the H6 heading 1`] = `
{
  "activate": true,
  "attributes": {
    "title": "Heading 6",
  },
  "category": "Generic blocks",
  "content": "
    <mj-text font-size="0.67rem" line-height="1.15">
      Heading
    </mj-text>",
  "label": "H6",
  "media": "
<svg
  class="lucide lucide-heading-6"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 12h8" />
  <path d="M4 18V6" />
  <path d="M12 18V6" />
  <circle cx="19" cy="16" r="2" />
  <path d="M20 10c-2 2-3 3.5-3 6" />
</svg>
",
}
`;
